src/main/java/com/autotune/analyzer/adapters/MetricMetadataAdapter.java

```diff
@@ -0,0 +1,77 @@
+package com.autotune.analyzer.adapters;
+
+import com.autotune.analyzer.utils.AnalyzerConstants;
+import com.autotune.common.data.metrics.AcceleratorMetricMetadata;
+import com.autotune.common.data.metrics.MetricMetadata;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import com.autotune.utils.KruizeConstants;
+
+
+import java.io.IOException;
+
+/**
+ * This adapter actually specifies the GSON to identify the type of implementation of MetricMetadata
```

Comment:
I think this javadoc needs to be more descriptive about current implementation and future prospects